src/plugins/discover/public/application/components/sidebar/discover_field_details.tsx

```diff
@@ -39,6 +39,16 @@ export function DiscoverFieldDetails({
   onAddFilter,
 }: DiscoverFieldDetailsProps) {
   const warnings = getWarnings(field);
+  const [showVisualizeLink, setShowVisualizeLink] = useState<boolean | undefined>(false);
+
+  isFieldVisualizable(field, indexPattern.id).then(
+    (flag) => {
+      setShowVisualizeLink(flag);
+    },
+    () => {
+      setShowVisualizeLink(false);
+    }
+  );
```

Comment:
`useEffect` should be used here, else the function is executed every time the Component is re-rendered